ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -0,0 +1,243 @@
+//
+//  LogFilesViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 15.05.2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2018, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+import ownCloudSDK
+
+class LogFileTableViewCell : ThemeTableViewCell {
+
+	static let identifier = "LogFileTableViewCell"
+
+	var shareAction : ((_ cell:UITableViewCell) -> Void)?
+
+	override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+		super.init(style: .subtitle, reuseIdentifier: reuseIdentifier)
+
+		let shareButton = UIButton(type: .system)
+		shareButton.setImage(UIImage(named: "open-in"), for: .normal)
+		shareButton.addTarget(self, action: #selector(shareButtonTapped), for: .touchUpInside)
+		shareButton.frame = CGRect(origin: CGPoint(x:0.0, y:0.0), size: shareButton.imageView!.image!.size)
```

Comment:
This will crash if there's no image set (i.e. when there's no longer an image named `open-in` bundled in the app). Maybe wrap the whole shareButton segment inside `if let image = UIImage(named: "open-in") {}`?

Reply:
@felix-schwarz for me it is a philosophical question: let the app crash because there is some problem and find the crash early in the QA process or may be ship not crashing app accidentally slipping through QA but containing visual glitches.